kernel/kernel_drivers.go

```diff
@@ -40,6 +40,12 @@ var osSymlink = os.Symlink
 // separated by dots for the kernel version.
 var utsRelease = regexp.MustCompile(`^([0-9]+\.){2}[0-9]+`)
 
+// EarlyKernelMountDir returns the mount directory for early on boot mount of
+// kernel snaps.
+func EarlyKernelMountDir(ksnapName string, rev snap.Revision) string {
+	return filepath.Join("/run/mnt/kernel-snaps", ksnapName, rev.String())
```

Comment:
should this perhaps use the root dir for easier testing?

```suggestion
	return filepath.Join(dirs.GlobalRootDir, "/run/mnt/kernel-snaps", ksnapName, rev.String())
```

Reply:
yes, usually we do that

Reply:
I did not want to do that as the path goes to the `where` in the mount unit, so then the file name for the unit needs to match and actually that provokes problems with unit tests. This might be workable, but requires changes in a few tests involving mount units, and we need also to remove things like the call to `StripRootDir` in https://github.com/snapcore/snapd/blob/master/systemd/systemd.go#L1561 .

Reply:
Hmm that piece in systemd package is a bit overeager to remove the root and could be masking some other issue, but perhaps saves us doing some test changes. I was looking at this part of code https://github.com/snapcore/snapd/blob/master/overlord/snapstate/backend/mountunit.go#L30-L31 where the root dir is stripped when passing the mount source and target, thus the mount units have the right content and names, the only relevant change being that they are created under the test root dir.

Reply:
I think we should do the consistent thing. We might need to add some more Strip... in systemd though.

Reply:
I've changed this now, indeed for the moment the price to pay is some calls to `dirs.StripRootDir`